templates/usr/local/bin/rsync_debian_mirror.j2

```diff
@@ -7,22 +7,24 @@
 # uncomment to enable bash debugging
 #set -x
 
-tmp_path="{{ repo_mirror_tmp_path }}"
-base_path="{{ repo_mirror_base_path }}/{{ item.name }}"
-log_file="{{ repo_mirror_log_path }}/{{ item.name }}/repo_{{ item.name }}.log"
-status_file="{{ repo_mirror_base_path }}/{{ item.name }}.json"
+repo_name={{ item.name | quote }}
+tmp_path={{ repo_mirror_tmp_path | quote }}
+base_path={{ repo_mirror_base_path | quote }}/$repo_name
+log_file={{ repo_mirror_log_path | quote }}/$repo_name/repo_$repo_name.log
+status_file=$base_path.json
+owner={{ repo_mirror_user | quote }}
 
-repo_source="{{ item.source_repo }}"
+repo_source={{ item.source_repo | quote }}
 
-datetime_format="{{ repo_mirror_datetime_format }}"
-bwlimit="{{ item.bwlimit | default(repo_mirror_bwlimit) }}"
-rsync_timeout="{{ item.rsync_timeout | default(repo_mirror_rsync_timeout) }}"
+datetime_format={{ repo_mirror_datetime_format | quote }}
+bwlimit={{ item.bwlimit | default(repo_mirror_bwlimit) | quote }}
+rsync_timeout={{ item.rsync_timeout | default(repo_mirror_rsync_timeout) | quote }}
 
-lock="/tmp/sync_{{ item.name }}_mirror.lck"
-rsyncexitcode="-1"
+lock=/tmp/sync_${repo_name}_mirror.lck
+rsyncexitcode=-1
 
-archexclude="{{ item.excludes | join(' ') | default("") }}"
-trace="project/trace/{{ repo_mirror_fqdn }}"
+archexclude={{ item.excludes | join(" ") | default("") | quote }}
```

Comment:
I haven't changed the order of this pipeline, but I don't entirely understand the logic of joining first and *then* using a default. Semantically, the following seems more understandable:

    {{ item.excludes | default([]) | join(" ") | … }}

But potentially I'm missing something here. Please let me know if you want to keep this as-is or change it.

Reply:
I think both are fine. but default first makes more sense

Reply:
I will switch it then (probably in a third commit, to avoid mixing things up).